src/plotman/manager.py

```diff
@@ -53,25 +53,39 @@ def phases_permit_new_job(phases: typing.List[job.Phase], d: str, sched_cfg: plo
     if len(phases) == 0:
         return True
 
-    milestone = job.Phase(
-        major=sched_cfg.tmpdir_stagger_phase_major,
-        minor=sched_cfg.tmpdir_stagger_phase_minor,
-    )
+    # Assign variables
+    major = sched_cfg.tmpdir_stagger_phase_major
+    minor = sched_cfg.tmpdir_stagger_phase_minor
     # tmpdir_stagger_phase_limit default is 1, as declared in configuration.py
-    if len([p for p in phases if p < milestone]) >= sched_cfg.tmpdir_stagger_phase_limit:
-        return False
-
-    # Limit the total number of jobs per tmp dir. Default to the overall max
+    stagger_phase_limit = sched_cfg.tmpdir_stagger_phase_limit
+    
+    # Limit the total number of jobs per tmp dir. Default to overall max
     # jobs configuration, but restrict to any configured overrides.
     max_plots = sched_cfg.tmpdir_max_jobs
-    if dir_cfg.tmp_overrides is not None and d in dir_cfg.tmp_overrides:
-        curr_overrides = dir_cfg.tmp_overrides[d]
+    
+    # Check if any overrides exist for the current job
+    if sched_cfg.tmp_overrides is not None and d in sched_cfg.tmp_overrides:
+        curr_overrides = sched_cfg.tmp_overrides[d]
+    
+        # Check for and assign major & minor phase overrides
+        if curr_overrides.tmpdir_stagger_phase_major is not None:
+            major = curr_overrides.tmpdir_stagger_phase_major
+        if curr_overrides.tmpdir_stagger_phase_minor is not None:
+            minor = curr_overrides.tmpdir_stagger_phase_minor
+        # Check for and assign stagger phase limit override
+        if curr_overrides.tmpdir_stagger_phase_limit is not None:
+            stagger_phase_limit = curr_overrides.tmpdir_stagger_phase_limit
+        # Check for and assign stagger phase limit override
         if curr_overrides.tmpdir_max_jobs is not None:
             max_plots = curr_overrides.tmpdir_max_jobs
-    if len(phases) >= max_plots:
+        
+    milestone = job.Phase(major,minor,)
+    
+    # Check if phases pass the criteria
+    if len([p for p in phases if p < milestone]) >= stagger_phase_limit or len(phases) >= max_plots:
```

Comment:
I don't have a formal argument for this, but I think I would rather keep this more in the form it was before.

```suggestion
    if len([p for p in phases if p < milestone]) >= stagger_phase_limit
        return False

    if len(phases) >= max_plots:
```

Reply:
No worries. I will shift it back.
To my eye, it looked neater and was easier to understand with less repetition